### PR TITLE
Add support for conditional breakpoints

### PR DESCRIPTION
**Pull request by thegecko**
PR to add support for conditional breakpoints.

Also includes a fix to ensure locations for break-inserts are correctly quoted.

### CODE REVIEW

**Comment by dschaefer** [comment]
Is there any way we can get unit tests for this feature. I'd kick off the tests but that won't tell us whether this works or not.

**Comment by thegecko** [comment]
All of our development is done using remote debugging, so I'd have to recreate this in a local environment in order to add tests. I may have some to work on it next week.

**Comment by thegecko** [comment]
I've added some breakpoint tests including a conditional one for this functionality.

File: src/integration-tests/test-programs/Makefile [hunk 1]
```diff
@@ -8,6 +8,9 @@ CXX = g++
 LINK = $(CC) -o $@ $^
 LINK_CXX = $(CXX) -o $@ $^
 
+loop: count.o
```
Comment (marechal-p) [inline src/integration-tests/test-programs/Makefile]:
```makefile
count: count.o // ?
```

Comment (thegecko) [reply src/integration-tests/test-programs/Makefile]: Yup, copy-pasta error, thanks.

**Comment by thegecko** [comment]
> I have build errors on my side:

Fixed, that's what comes from working on a mac but having to copy-paste to a linux vm to run the tests :/

File: src/integration-tests/breakpoints.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        const vr = scope.scopes.body.scopes[0].variablesReference;
+        const vars = await dc.variablesRequest({ variablesReference: vr });
+        verifyVariable(vars.body.variables[0], 'count', 'int', '0');
+    });
+
+    it('hits a conditional breakpoint', async () => {
+        await dc.setBreakpointsRequest({
+            source: {
+                name: 'count.c',
+                path: '/home/rob/cdt-gdb-adapter/src/integration-tests/test-programs/count.c',
```
Comment (marechal-p) [inline src/integration-tests/breakpoints.spec.ts]: Absolute path should be relative, see https://github.com/eclipse-cdt/cdt-gdb-adapter/blob/master/src/integration-tests/var.spec.ts#L25

Comment (thegecko) [reply src/integration-tests/breakpoints.spec.ts]: D'oh, thanks!

Comment (thegecko) [reply src/integration-tests/breakpoints.spec.ts]: Fixed

**Comment by marechal-p** [comment]
Please add `count` to `src/integration-tests/test-programs/.gitignore`, just noticed my index getting dirty because of this :)

**Comment by thegecko** [comment]
@marechal-p added

**Pull request merged**